packages/nitro-protocol/src/contract/outcome.ts

```diff
@@ -1,181 +1,67 @@
-// As the Outcome struct defined in Outcome.sol
+// CONSTANTS
+export const MAX_OUTCOME_ITEMS = 600;
 
 export function hashOutcome(outcome: Outcome): Bytes32 {
   const encodedOutcome = encodeOutcome(outcome);
   return utils.keccak256(encodedOutcome);
 }
 
-export function decodeOutcome(encodedOutcome: Bytes): Outcome {
-  const assetOutcomes = utils.defaultAbiCoder.decode(
-    ['tuple(address asset, bytes outcomeContent)[]'],
-    encodedOutcome
-  )[0];
-  return assetOutcomes.map((a: AssetOutcome & {outcomeContent: string}) =>
-    decodeOutcomeItem(a.outcomeContent, a.asset)
-  );
-}
-
-export function encodeOutcome(outcome: Outcome): Bytes32 {
-  const encodedAssetOutcomes = outcome.map((o: AssetOutcome) => {
-    let encodedData;
-    let outcomeType;
-    if (isAllocationOutcome(o)) {
-      encodedData = encodeAllocation(o.allocationItems);
-      outcomeType = AssetOutcomeType.AllocationOutcomeType;
-    } else {
-      encodedData = encodeGuarantee(o.guarantee);
-      outcomeType = AssetOutcomeType.GuaranteeOutcomeType;
-    }
-
-    return {
-      asset: o.asset,
-      outcomeContent: encodeAssetOutcomeFromBytes(outcomeType, encodedData),
-    };
-  });
-
-  return utils.defaultAbiCoder.encode(
-    ['tuple(address asset, bytes outcomeContent)[]'],
-    [encodedAssetOutcomes]
-  );
+export type SimpleAllocation = ExitFormat.Allocation & {
+  allocationType: ExitFormat.AllocationType.simple;
+};
+
+export type GuaranteeAllocation = ExitFormat.Allocation & {
+  // this will cause executeExit to revert, which is what we want for a guarantee
+  // it should only work with a custom 'claim' operation
+  allocationType: ExitFormat.AllocationType.guarantee;
+};
+
+export type SingleAssetGuaranteeOutcome = ExitFormat.SingleAssetExit & {
+  allocations: GuaranteeAllocation[];
+};
+
+export type GuaranteeOutcome = SingleAssetGuaranteeOutcome[];
+const A_ADDRESS = '0x00000000000000000000000096f7123E3A80C9813eF50213ADEd0e4511CB820f';
+const B_ADDRESS = '0x00000000000000000000000053484E75151D07FfD885159d4CF014B874cd2810';
+const exampleGuaranteeOutcome1: GuaranteeOutcome = [
+  {
+    asset: constants.AddressZero,
+    metadata: '0x',
+    allocations: [
+      {
+        destination: '0xjointchannel1',
+        amount: '0xa',
+        allocationType: ExitFormat.AllocationType.guarantee,
+        metadata: encodeGuaranteeData([B_ADDRESS, A_ADDRESS]),
+      },
+      {
+        destination: '0xjointchannel2',
+        amount: '0xa',
+        allocationType: ExitFormat.AllocationType.guarantee,
+        metadata: encodeGuaranteeData([A_ADDRESS, B_ADDRESS]),
+      },
+    ],
+  },
+];
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const exampleGuaranteeOutcome2: ExitFormat.Exit = exampleGuaranteeOutcome1; // GuaranteeOutcome is assignable to Exit
+
+export function encodeGuaranteeData(destinations: string[]): BytesLike {
+  return defaultAbiCoder.encode(['bytes32[]'], [destinations]);
```

Comment:
```suggestion
  return defaultAbiCoder.encode(['bytes32[]'], destinations);
```
:question: 

`encode` here wants an array as its second parameter, but `destinations` as passed in is already an array. This double-wrapped array seems intentional, based on `decodeGuaranteeData` function, which returns the first element of the decoded data.

Just pointing at it because it pops out as a potential typo style error.

Reply:
I think this is just the [ethers API](https://docs.ethers.io/v5/api/utils/abi/coder/#AbiCoder-encode) which calls for an array of types and an array of values. 

The `decode()[0]` returning the first element is (I think) also just a quirk of the ethers API, which returns the decoded data inside of an array.

Reply:
On https://playground.ethers.org/: 

```
utils.defaultAbiCoder.encode(['bytes32[]'], [[constants.HashZero]]);
"0x000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000"
utils.defaultAbiCoder.encode(['bytes32[]'], [constants.HashZero]);
expected array value (argument=null, value="0x0000000000000000000000000000000000000000000000000000000000000000", code=INVALID_ARGUMENT, version=abi/5.4.0)
```